### ISSUE TITLE
Home Group UI

### ISSUE THREAD

**Issue by PROgram52bc** [corpus-christi/corpus-christi#542]
Database and API endpoints related to the new `Group` module are mostly done. The goal of this issue is to implement the client-side management interface for the `Group` module.
Specifics are identified in [the documentation](https://github.com/corpus-christi/corpus-christi/blob/development/doc/stories.md#group-admin).

**Comment by quiteqiang** [comment]
HomeGroup UI:
- [x] Add Manager table
- [x] Add Group-type to group list (name, description, manager)
- [x] Add Person button is not working

Message requirement:
- [x] Manager can elect to have the “From” or “Reply-To” fields of an outbound email set either to their own address or to the generic system one.

Admin Tree View:

- [x] Add send email UI (tree view) (In process)
- [x] Make tree-view of managers and members
- [x] Implement a utility function to `getSubGroups(group, groups, subGroups)` of a person

  asthetics
  - [x] use different icons for different types of objects
- [x] implement the alternative hierarchical treeview?
- [x] add buttons to "show managers" or "show members"
- [x] split "managers" to different manager types

Permission:

- [x] Add a person: shouldn't be given any authority to add/delete/promote at the first place
- [x] As a normal member, he/she shouldn't see the functional buttons (other buttons)
- [x] Entity search should search for only unarchived entities

Enhancement:
- [x] Line graph for both Group(Overseers & Leaders )
Requirements for the Line Graph:
- [x] Monthly 

>  - [x] Be able to load multiple groups and show their graphs corresponding to the data from the database
>  - [x] Be able to select more than one groups on  the chart
>  - [x] The graph generator should be given each graph a different color & tag & data 

- [x] Weekly 

> - [x] Show every 4 weeks in a month of each month
> - [x] automatically switch to the previous/following year one the user wants to go to the previous year
> - [x] Use another way to load data from the database
> - [x] Once the page is switched to "Weekly," the time scale of the graph should be switched to another scale

- [x] Annually  

> - [x] This should be 2 years/3 years in a roll
> - [x] Calculate the data in a new way.
> - [x] The year should be switched after clicking the "Left" and "Right" button on the  
> - [x] Once the page is switched to "Annually," the time scale of the graph should be switched to another scale

- [x] In Detail filed, "Group Description" is not assigned/properly showed (need change if necessary)


<img width="885" alt="Screen Shot 2020-06-10 at 5 53 00 PM" src="https://user-images.githubusercontent.com/47837723/84322592-3ba44000-ab43-11ea-896c-c6a89fb3a7bb.png">

CRUD functionality:
- [x] Create Group function: change "Search Managers" to "Search Group Type."
- [x] Show managers
- [x] Archive/unarchive meeting functionality
- [x] Edit meeting functionality
- [x] Take attendance on a meeting
- [x] Add a person as a normal Group member
- [x] Show "archived" filter menu on meetings
- [x] Show "attendance count" on the meeting list
- [x] Reusable component to create a group-type, should be able to plug-and-use in "create group" and "create a group type" view.
- [x] Reusable component to create a manager-type.
- [x] Members list add a button to 'promote as manager', show a dialog, being able to select manager type and create a manager type
- [x] Managers list add a button to 'demote as a member,' remove the archive button.

People UI:
- [x] "Add Person" button is not functioning correctly

**Comment by PROgram52bc** [comment]
Optimizations:

Treeview:
- [ ] auto-expand all when treeview is loaded (may need to modify the computed property to a function that is called on page load, and expand the tree once the function is complete)
- [x] add checkboxes to check all managers/members, remove the showManagers/members selection.
- [ ] split managers into manager types

manager&member list:
- [x] add loading state for unarchive button

**Comment by quiteqiang** [comment]
Line Graph issue:
- [x] When the mouse hovers over any of the labels of a graph, the previous graph flash in and out all the time.  Need to clean the data before drawing a new graph

**Comment by quiteqiang** [comment]
Unsolved issues are added into new issue:  [https://github.com/corpus-christi/corpus-christi/issues/619](url)
